bciers/apps/reporting/tests/components/ActivityForm.test.tsx

```diff
@@ -0,0 +1,128 @@
+import { render, screen } from "@testing-library/react";
+import { describe, expect, vi, it, beforeEach } from "vitest";
+import ActivityForm from "@reporting/src/app/components/activities/ActivityForm";
+import { actionHandler } from "@bciers/testConfig/mocks";
+
+// Mock data
+const mockActivityData = {
+  activityId: 1,
+  sourceTypeMap: {
+    1: "firstTestSourceType",
+    2: "secondTestSourceType",
+  },
+};
+
+const mockReportDate = "2024-01-01";
+
+const mockUiSchema = {
+  "ui:classNames": "form-heading-label",
+  firstTestSourceType: {
+    "ui:widget": "CheckboxWidget",
+    "ui:options": {
+      label: false,
+    },
+  },
+  secondTestSourceType: {
+    "ui:widget": "CheckboxWidget",
+    "ui:options": {
+      label: false,
+    },
+  },
+};
+
+const mockdefaultSourceType = { units: [{ fuels: [{ emissions: [{}] }] }] };
+
+const response = {
+  schema: {
+    type: "object",
+    title: "General stationary combustion",
+    properties: {
+      firstTestSourceType: {
+        type: "boolean",
+        title: "First Test Source Type Title",
+      },
+      secondTestSourceType: {
+        type: "boolean",
+        title: "Second Title",
+      },
+    },
+  },
+};
+
+const flushPromises = () => new Promise((resolve) => setTimeout(resolve, 0));
```

Comment:
This looks like something `vi.runAllTicks` or `runAllTimers` would do for us https://vitest.dev/api/vi.html#vi-runallticks

Reply:
Doesn't appear to work the same. Tests still get stuck in the ...Loading state